Project_source/app/src/main/java/android/clicker/school_live_simulator/BadState.kt

```diff
@@ -0,0 +1,9 @@
+package android.clicker.school_live_simulator
+
+class BadState : PlayerState() {
+    init {
```

Comment:
You don't need init block here. Remove it in every class

Reply:
How I should initialize values in subclasses?

Reply:
You can use open protected + override but now I think that your solution is better. @Ferrociraptus, what do you like more?

Reply:
I think, in abstract class PlayerState fields should be public, abstract and immutable(val), for example 
abstract val reduce_school_performance_percent: Int
The advantages are:
- fields are safe because val-properties can't be changed
- we can easily get values because they are public
- In subclasses we just override fields with necessary values, so we don't have to use init blocs or something else. For example: override val reduce_school_performance_percent: Int = 0

Reply:
@Bazalii, I think, that @Endray952 says true things. The best way it is abstract public val fields.

Reply:
@Endray952 yes, use abstract public val without values in abstract class. Then override each one with correct values